### PR TITLE
Use setItems instead of setDataSource

### PR DESCRIPTION
**Pull request by pleku**
Part of vaadin/flow#8632

### CODE REVIEW

**Comment by pleku** [comment]
DO NOT MERGE (for now)

File: vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java [hunk 1]
```diff
@@ -279,21 +286,135 @@ public Registration addCollapseListener(
...
+     * @return the data view
+     * @deprecated use {@link #setDataProvider(HierarchicalDataProvider)},
+     *             {@link #setItems(Collection, ValueProvider)},
+     *             {@link #setItems(Stream, ValueProvider)} or
+     *             {@link #setTreeData(TreeData)} instead.
+     */
+    @Deprecated
+    @Override
+    public GridListDataView<T> setItems(Collection<T> items) {
+        return null;
```
Comment (mshabarov) [inline vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java:417]: `throw new IllegalArgumentException` is missing

File: vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java [hunk 2]
```diff
@@ -780,10 +780,10 @@ private void createGridWithExactRowCount() {
 
         /*
          * In case it is desired to show to the user the exact number of rows in
-         * the data source, that can be done providing another callback that
-         * fetches the row count from the data source.
+         * the backend, that can be done providing another callback that
+         * fetches the row count from the backend.
          */
-        GridLazyDataView<Person> lazyDataView = grid.setDataSource(
+        GridLazyDataView<Person> lazyDataView = grid.setItems(
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java:786]: It is not deprecated ???

File: vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java [hunk 3]
```diff
@@ -748,15 +748,15 @@ private void createGridWithCustomRowCountEstimate() {
         PersonService personService = new PersonService();
 
         GridLazyDataView<Person> lazyDataView = grid
-                .setDataSource(query -> personService
+                .setItems(query -> personService
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java:751]: It is not deprecated ???

File: vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java [hunk 4]
```diff
@@ -726,9 +726,9 @@ private void createGridWithLazyLoading() {
          * based on offset and limit.
          *
          * When the user scrolls to the end grid will automatically extend and
-         * fetch more rows until the data source runs out of items.
+         * fetch more rows until the backend runs out of items.
          */
-        grid.setDataSource(query -> personService
+        grid.setItems(query -> personService
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java:731]: It is not deprecated ???

**Pull request merged**